### ISSUE TITLE
Stream cut for a few seconds

### ISSUE THREAD

**Issue by northerndub** [torch2424/live-stream-radio#104]
I'm on Windows and streaming to youtube, after every song the stream is down for like 1-2 seconds. Is there any way to fix this or is my setup to slow to get the next song going without a cut?

**Comment by torch2424** [comment]
Yup perfect response! I want to work on a "single video" mode at somepoint, so this doesn't happen, but that would be way far down the line :)

**Comment by torch2424** [comment]
Are you using gifs by any chance? MP4s load better. But yes, no matter what, there will be cuts in the stream. Perhaps I can make it configurable to make the cuts have empty audio before and after? 😄

**Comment by AndreasWebdev** [comment]
You can also try to disable the latency-free mode on youtube's dashboard to get a 30 seconds delay as a buffer.

**Comment by northerndub** [comment]
I did try mp4 and gifs and the difference is not significant. The delay only delays the cut.

**Comment by torch2424** [comment]
I would try what @AndreasWebdev I personally use that on my Youtube streams in the past, and had no issues 😮

**Comment by AndreasWebdev** [comment]
Yeah my DigitalOcean always catches up the second of delay.. could you post the config file (without your stream key) in here? @northerndub

**Comment by northerndub** [comment]
{
  "stream_url": "rtmp://a.rtmp.youtube.com/live2/$stream_key",
  "stream_key": "",
  "ffmpeg_path": "",
  "video_codec": "libx264",
  "audio_codec": "aac",
  "audio_bit_rate": "128k",
  "audio_sample_rate": "44100",
  "normalize_audio": true,
  "video_height": "480",
  "video_width": "854",
  "video_bit_rate": "500k",
  "video_fps": "12",
  "bufsize": "512k",
  "crf": "28",
  "preset": "veryfast",
  "threads": 6,
  "max_gif_size": "720",
  "api": {
    "host": "localhost",
    "port": "8000",
    "key": "super-secret-api-key",
    "number_of_history_items": 100
  },
  "radio": {
    "audio_directory": "./audio",
    "video_directory": "./video",
    "overlay": {
      "enabled": true,
      "font_path": "./fonts/Lato-Regular.ttf",
      "title": {
        "enabled": true,
        "text": "STEPPAS DUB & REGGAE 24/7 Livestream Radio",
        "font_color": "#FFFFFF",
        "font_border": "#000000",
        "font_size": "10",
        "enable_scroll": true,
        "font_scroll_speed": "20",
        "position_x": "0",
        "position_y": "5"
      },
      "artist": {
        "enabled": true,
        "label": "Artist: ",
        "font_color": "#FFFFFF",
        "font_border": "#000000",
        "font_size": "10",
        "position_x": "2",
        "position_y": "15"
      },
      "album": {
        "enabled": false,
        "label": "Album: ",
        "font_color": "#FFFFFF",
        "font_border": "#000000",
        "font_size": "10",
        "position_x": "2",
        "position_y": "25"
      },
      "song": {
        "enabled": true,
        "label": "Song: ",
        "font_color": "#FFFFFF",
        "font_border": "#000000",
        "font_size": "10",
        "position_x": "2",
        "position_y": "25"
      },
      "image": {
        "enabled": false,
        "image_path": "./live-stream-radio-overlay-image.png",
        "position_x": 0,
        "position_y": 0
      }
    }
  },
  "interlude": {
    "enabled": false,
    "frequency": "0.2",
    "audio_directory": "./interludes/audio",
    "video_directory": "./interludes/video",
    "overlay": {
      "enabled": false,
      "font_path": "./fonts/Lato-Regular.ttf",
      "title": {
        "enabled": false,
        "text": "Please wait, music shall resume shortly...",
        "font_color": "#FFFFFF",
        "font_border": "#000000",
        "font_size": "10",
        "position_x": "2",
        "position_y": "5"
      },
      "image": {
        "enabled": false,
        "image_path": "./live-stream-radio-overlay-image.png",
        "position_x": 0,
        "position_y": 0
      }
    }
  }
}

**Comment by torch2424** [comment]
@AndreasWebdev the config looks fine, and isn't using any high performance values.

@northerndub Yeah, honestly, this is how live-stream-radio works. I tried looking into how to keep it continuous, but I could not find a way to have an ffmpeg process switch inputs on the fly. I'll keep this open, and feel free to suggest any solutions to this in the code.

**Comment by AndreasWebdev** [comment]
I've also tried researching it, in the end, programs like OBS can do it as well, so there is got to be a way (though they might have an active video stream that they can manipulate internally)

**Comment by torch2424** [comment]
I think what OBS does is take in VLC as an input. And VLC manages that for them? 

I can definitely look into that, though, I wouldn't want to add VLC as a dependency

**Comment by StoneCypher** [comment]
Here are some approaches to switching ffmpeg inputs on the fly, which is the correct solution layer.

[Hot junction .3gp](https://unix.stackexchange.com/questions/454580/ffmpeg-change-input-without-stoping-proccess)

[Use HTTP Live Streaming](https://stackoverflow.com/questions/42737129/change-ffmpeg-input-on-the-fly)

[Multiplex through mpeg-ts](https://stackoverflow.com/questions/32972903/switch-between-multiple-stream-inputs-with-nginx-rtmp)

**Comment by StoneCypher** [comment]
You can also do it manually with [streamselect](https://ffmpeg.org/ffmpeg-filters.html#streamselect_002c-astreamselect) but that's pretty difficult

**Comment by torch2424** [comment]
@StoneCypher Woah! Dude I've been looking for a solution for this for like forever, I am soooo stoked you found these!

Keeping it real, I have been super busy in other projects involving WebAssembly. Thus, I'd honestly try these out because they seem like they would work. Particularly, the HLS looks the easiest / simplest to do. But I doubt I'd have the time until I get some vacation days from my new job that I just started like a month ago 😂 

@StoneCypher could you open a PR for this? If so, I'd totally take the time to try it out and merge it 😄 

Also, I saw on your Github profile you're based out of SF? I'll be living out here for a couple more weeks. If you want to chat about the project, we can grab coffee some time! Just hit me up on Twitter or something 😂

**Comment by beamercola** [comment]
I'm streaming this through Ant media server and playing it on iOS.  It seems to issue a stop command on AVPlayer while it's changing tracks.  This library is so incredible btw.

**Comment by torch2424** [comment]
@beamercola Thanks for the feedback! I super appreciate it! 😄 👍
